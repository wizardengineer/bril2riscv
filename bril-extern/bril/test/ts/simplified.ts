erronious(1n);

// This snippet checks that the ts compiler doesn't try to create an unreachable else branch
function erronious(x: bigint): bigint {
    if (true) {
        return x;
    }
}
