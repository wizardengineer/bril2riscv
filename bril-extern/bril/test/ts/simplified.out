@main {
  v0: int = const 1;
  call @erronious v0;
  v1: int = const 0;
}
@erronious(x: int): int {
  v1: bool = const true;
  br v1 .then.0 .else.0;
.then.0:
  v2: int = id x;
  ret v2;
.else.0:
}
